dcp_prototype/backend/wrangling/migrations/scripts/migrate_dcp_one_to_dcp_two.py

```diff
@@ -167,185 +184,159 @@ def generate_metadata_structure_from_s3_uri(s3_uri, num_threads):
-
-        # Find a SequenceFile object with similar filename and update it with the S3 URI
-        # if it is a FASTQ file
-        for sequence_file_id, sequence_file in dataset_metadata.sequence_files.items():
-            if sequence_file.filename == filename:
-                s3_uri = export_file_to_s3(source_location, filename)
-                sequence_file.s3_uri = s3_uri
-                dataset_metadata.sequence_files[sequence_file_id] = sequence_file
-                continue
-
-        # Find an AnalysisFile object with similar filename and update it with the S3
-        # URI if it is not a FASTQ or a JSON file (likely a Zarr or BAM file).
-        for analysis_file_id, analysis_file in dataset_metadata.analysis_files.items():
-            if analysis_file.filename == filename:
-                s3_uri = export_file_to_s3(source_location, filename)
-                sequence_file.s3_uri = s3_uri
-                dataset_metadata.analysis_files[analysis_file_id] = analysis_file
-                continue
-
-    print(f"Completed uploading all files to S3")
-
-
-def export_file_to_s3(full_source_file_path, filename):
-    s3_uri = f"s3://{BUCKET_NAME}/{PREFIX}/{filename}"
-
-    try:
-        S3_CLIENT.head_object(Bucket=BUCKET_NAME, Key=PREFIX + filename)
-        print(f"Found file {filename} so not re-uploading to S3.")
-        return s3_uri
-    except Exception:
-        pass
-
-    print(f"Uploading file {filename}")
-    try:
-        S3_CLIENT.upload_file(full_source_file_path, BUCKET_NAME, PREFIX + filename)
-    except ClientError as e:
-        print(e)
-        return False
-
-    return s3_uri
-
-
-if __name__ == "__main__":
+def main():
     parser = ArgumentParser()
     parser.add_argument(
         "-i",
-        "--input_directory",
-        nargs="+",
-        required=False,
-        help="A data directory contains all files that were a part of a single DCP 1.0 "
-        "project to be transformed into a valid DCP 2.0 metadata and inputted into"
-        " the DCP 2.0 ledger.",
-    )
-    parser.add_argument(
-        "-b",
-        "--bucket",
-        nargs="+",
-        required=False,
-        help="The bucket into which files should be transfered if transferring local "
-        "files as part of the project's metadata migration.",
-    )
-    parser.add_argument(
-        "-p",
-        "--prefix",
-        nargs="+",
-        required=False,
-        help="The bucket's prefix into which files should be transfered if transferring"
-        " local files as part of the project's metadata migration.",
+        "--input-source",
+        required=True,
+        help="input_source contains all metadata files that were part of a single DCP 1.0 project."
+        " It may be one of the following:  An s3 bucket containing json files."
+        " A directory containing json files."
+        " An s3 object in the tar.gz format containing the json files."
+        " Or a local tar.gz file containing the json files.",
```

Comment:
Below, it says that local directories are not supported?

Reply:
Do you mean this part:
```
        if s3_uri is None:
            print("You must specify an s3uri")
            sys.exit(1)
```

The OldDatasetMetadata takes in an s3uri as a constructor argument.  If that is not provided you run into an error later on.   I'm not sure what s3uri is supposed to be set to, so I plum that through as a command line argument. 

The code should work for local directories.

Reply:
Ahh s3 uri must the location where to put the files because the file location in s3 is included as part of the metadata.